pkg/util/hardware/mem_info_darwin.go

```diff
@@ -0,0 +1,52 @@
+//go:build darwin || freebsd || openbsd
+// +build darwin freebsd openbsd
+
+package hardware
+
+import (
+	"github.com/milvus-io/milvus/pkg/log"
+	"github.com/shirou/gopsutil/v3/mem"
+	"go.uber.org/zap"
+)
+
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// GetUsedMemoryCount returns the memory usage in bytes.
+func GetUsedMemoryCount() uint64 {
+	icOnce.Do(func() {
+		ic, icErr = inContainer()
+	})
+	if icErr != nil {
+		log.Error(icErr.Error())
+		return 0
+	}
+	if ic {
```

Comment:
do we still need to care about in container or not?

Reply:
the `gopsutil`'s 'VirtualMemory()` returns the system's memory usage even we run Milvus in container for some version. Only keeps this for only Darwin/FreeBSD/OpenBSD, as the new method doesn't work for them